### DISCUSSION TITLE
How are sensor names mapped from modbus.yaml to config.yaml?

### DISCUSSION THREAD

**Discussion by slee77** [StealthChesnut/HA-FoxESS-Modbus#20] [Q&A] [answered]
As per the title really.

I'm looking to tinker with what is possible with the sensor data from my FoxEss inverter.

I am unclear as to how the sensor names are converted from being space or hyphen separated to being snake case in the config YAML?

For example:
Grid CT  [modbus.yaml]   ->  sensor.grid_ct  [config.yaml] 
Battery-Discharge-Power [modbus.yaml] ->  [config.yaml] sensor.battery_discharge_power

I'm not sure if I'm missing a step in the translation between sensor data and config, or whether the translation is implicit in Python or whatever languague this is using under the hood?

Thanks for any pointers you can throw my way!

**Comment by Daibutt** [comment, marked as answer]
AFAIK it's all implicit within HA. In modbus.yaml there is a sensors section, and each sensor has a name. HA creates a sensor in the format "sensor.name". 
I'm guessing the underlying code/language isn't keen on spaces, -,  or mixed case so a sensor name of "My Best Sensor" creates the sensor "sensor.my_best_sensor".
This isn't unique to this integration, it's just a HA thing, or at least that's my understanding.

**Reply by slee77** [reply]
Thanks. I won’t worry any more about it and tinker away! 🙂

**Reply by StealthChesnut** [reply]
@Daibutt has got it right.

The sensors section defines the sensors that HA creates. You can define "name: " and "friendly_name: " on sensors for the internal and UI-displayed sensor names, but I'll be honest I was lazy.

HA takes the name and (I think) replaces non-alphanumeric characters with underscores, so pv1-power becomes pv1_power.

Look forward to seeing what you do!
